atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -149,13 +174,58 @@ public void setup() {
     @Override
     public void tearDown() {
         // This is required because of JUnit memory issues
+        sweepQueue.close();
         sweepQueue = null;
         progress = null;
         sweepableTimestamps = null;
         sweepableCells = null;
         puncherStore = null;
     }
 
+    @Test
+    public void secondaryQueueReadsFirstQueueLastSweptTimestamp() {
+        MetricsManager secondQueueManager = MetricsManagers.createForTests();
+        TargetedSweeper secondQueue = TargetedSweeper.createUninitialized(
+                secondQueueManager, runtimeSupplier::get, installConfig, ImmutableList.of());
+        secondQueue.initializeWithoutRunning(
+                timestampsSupplier,
+                mock(TimelockService.class),
+                spiedKvs,
+                TransactionServices.createV1TransactionService(spiedKvs),
+                mock(TargetedSweepFollower.class));
+        secondQueue.runInBackground();
+        await(() -> assertThat(secondQueueManager).hasLastSweptTimestampConservativeEqualTo(-1L));
+
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS);
+        sweepNextBatchForShards(CONSERVATIVE, DEFAULT_SHARDS);
+        await(() -> assertThat(secondQueueManager).hasLastSweptTimestampConservativeEqualTo(maxTsForFinePartition(0)));
+    }
+
+    @Test
+    public void increaseInShardsReflectedOnLastSweptTimestamp() {
+        sweepQueue.runInBackground();
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS);
+        sweepNextBatchForShards(CONSERVATIVE, DEFAULT_SHARDS);
+        await(() -> assertThat(metricsManager).hasLastSweptTimestampConservativeEqualTo(maxTsForFinePartition(0)));
+
+        runtimeSupplier.set(ImmutableTargetedSweepRuntimeConfig.builder()
+                .from(runtimeSupplier.get())
+                .shards(DEFAULT_SHARDS + 1)
+                .build());
+
+        // Shard count is memoized, creating a second queue bypasses waiting for expiration
+        TargetedSweeper secondQueue = TargetedSweeper.createUninitialized(
+                metricsManager, runtimeSupplier::get, installConfig, ImmutableList.of());
+        secondQueue.initializeWithoutRunning(
+                timestampsSupplier,
+                mock(TimelockService.class),
```

Comment:
How come you can't reuse the existing timelockService and mockFollower?

Reply:
No reason actually - I was just being too defensive in how I wrote it (trying to make sure there is little unnecessary overlap between the two queues as we are testing that one reads the other's updates). It's better to just re-use as you said!